diff --git a/gettext-runtime/configure b/gettext-runtime/configure
index 2d1444b..960aa58 100755
--- a/gettext-runtime/configure
+++ b/gettext-runtime/configure
@@ -12268,6 +12268,10 @@ aix[4-9]*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+openedition)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 beos*)
   lt_cv_deplibs_check_method=pass_all
   ;;
diff --git a/gettext-runtime/intl/configure b/gettext-runtime/intl/configure
index bbd545e..dacb9e2 100755
--- a/gettext-runtime/intl/configure
+++ b/gettext-runtime/intl/configure
@@ -10408,6 +10408,10 @@ openbsd*)
   fi
   ;;
 
+openedition)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 osf3* | osf4* | osf5*)
   lt_cv_deplibs_check_method=pass_all
   ;;
diff --git a/gettext-runtime/libasprintf/configure b/gettext-runtime/libasprintf/configure
index b88175f..735b08b 100755
--- a/gettext-runtime/libasprintf/configure
+++ b/gettext-runtime/libasprintf/configure
@@ -10470,6 +10470,10 @@ openbsd*)
   fi
   ;;
 
+openedition)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 osf3* | osf4* | osf5*)
   lt_cv_deplibs_check_method=pass_all
   ;;
diff --git a/gettext-tools/configure b/gettext-tools/configure
index de76ed9..e53b443 100755
--- a/gettext-tools/configure
+++ b/gettext-tools/configure
@@ -16133,6 +16133,10 @@ aix[4-9]*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+openedition)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 beos*)
   lt_cv_deplibs_check_method=pass_all
   ;;
diff --git a/libtextstyle/configure b/libtextstyle/configure
index 8b2c75e..5b62e61 100755
--- a/libtextstyle/configure
+++ b/libtextstyle/configure
@@ -10856,6 +10856,10 @@ aix[4-9]*)
   lt_cv_deplibs_check_method=pass_all
   ;;
 
+openedition)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 beos*)
   lt_cv_deplibs_check_method=pass_all
   ;;
