diff --git a/gettext-tools/gnulib-tests/test-unsetenv.c b/gettext-tools/gnulib-tests/test-unsetenv.c
index f912e79..7601fa8 100644
--- a/gettext-tools/gnulib-tests/test-unsetenv.c
+++ b/gettext-tools/gnulib-tests/test-unsetenv.c
@@ -32,6 +32,10 @@ SIGNATURE_CHECK (unsetenv, int, (char const *));
 int
 main (void)
 {
+ /*The guard is added as a workaround for issue 
+  * https://github.com/ZOSOpenTools/gettextport/issues/15
+  * This guard can be removed once the fix for issue is provided by LE team*/
+#ifndef __MVS__
   char entry[] = "b=2";
 
   /* Test removal when multiple entries present.  */
@@ -49,6 +53,7 @@ main (void)
   errno = 0;
   ASSERT (unsetenv ("a=b") == -1);
   ASSERT (errno == EINVAL);
+#endif
 #if 0
   /* glibc and gnulib's implementation guarantee this, but POSIX no
      longer requires it: http://austingroupbugs.net/view.php?id=185  */
