diff --git a/gettext-tools/gnulib-tests/test-unsetenv.c b/gettext-tools/gnulib-tests/test-unsetenv.c
index 242d17e..cb11966 100644
--- a/gettext-tools/gnulib-tests/test-unsetenv.c
+++ b/gettext-tools/gnulib-tests/test-unsetenv.c
@@ -32,6 +32,10 @@ SIGNATURE_CHECK (unsetenv, int, (char const *));
 int
 main (void)
 {
+ /*The guard is added as a workaround for issue 
+  * https://github.com/ZOSOpenTools/gettextport/issues/15
+  * This guard can be removed once the fix for issue is provided by LE team*/
+#ifndef __MVS__
   /* Static to pacify gcc -Wanalyzer-putenv-of-auto-var.  */
   static char entry[] = "b=2";
 
@@ -50,6 +54,7 @@ main (void)
   errno = 0;
   ASSERT (unsetenv ("a=b") == -1);
   ASSERT (errno == EINVAL);
+#endif
 #if 0
   /* glibc and gnulib's implementation guarantee this, but POSIX no
      longer requires it: https://austingroupbugs.net/view.php?id=185  */
