diff --git a/gettext-tools/gnulib-tests/test-unsetenv.c b/gettext-tools/gnulib-tests/test-unsetenv.c
index d8e5b01..db92f13 100644
--- a/gettext-tools/gnulib-tests/test-unsetenv.c
+++ b/gettext-tools/gnulib-tests/test-unsetenv.c
@@ -34,7 +34,7 @@ main (void)
 {
   /* Static to pacify gcc -Wanalyzer-putenv-of-auto-var.  */
   static char entry[] = "b=2";
-
+#ifndef __MVS__
   /* Test removal when multiple entries present.  */
   ASSERT (putenv ((char *) "a=1") == 0);
   ASSERT (putenv (entry) == 0);
@@ -50,6 +50,7 @@ main (void)
   errno = 0;
   ASSERT (unsetenv ("a=b") == -1);
   ASSERT (errno == EINVAL);
+#endif
 #if 0
   /* glibc and gnulib's implementation guarantee this, but POSIX no
      longer requires it: http://austingroupbugs.net/view.php?id=185  */
