diff --git a/gettext-runtime/intl/gettextP.h b/gettext-runtime/intl/gettextP.h
index 1ac369e..e5ef60f 100644
--- a/gettext-runtime/intl/gettextP.h
+++ b/gettext-runtime/intl/gettextP.h
@@ -229,7 +229,7 @@ struct loaded_l10nfile *_nl_find_domain (const char *__dirname,
 #if defined _WIN32 && !defined __CYGWIN__
 					 const wchar_t *__wdirname,
 #endif
-					 char *__locale,
+					 char *__localename,
 					 const char *__domainname,
 					 struct binding *__domainbinding)
      attribute_hidden;
