diff --git a/gettext-runtime/intl/lock.c b/gettext-runtime/intl/lock.c
index f509e21..77996b9 100644
--- a/gettext-runtime/intl/lock.c
+++ b/gettext-runtime/intl/lock.c
@@ -20,5 +20,15 @@
 #include <config.h>
 
 #include "lock.h"
+#ifdef __MVS__
+  /*
+   * This pragma is required to avoid a collision with another 'lock' 
+   * code section and keep the z/OS binder happy
+   */
+  #pragma csect(code, "LOCKINTL#C")
+  #pragma csect(static, "LOCKINTL#S")
+#endif
 
 /* ========================================================================= */
 
