diff --git a/gettext-tools/autotools/cvsuser.c b/gettext-tools/autotools/cvsuser.c
index 22f5a50..1288dbc 100644
--- a/gettext-tools/autotools/cvsuser.c
+++ b/gettext-tools/autotools/cvsuser.c
@@ -50,7 +50,9 @@ struct passwd * getpwnam (const char *name)
   pw.pw_passwd = "*";
   pw.pw_uid = 100;
   pw.pw_gid = 100;
+#ifndef __MVS__
   pw.pw_gecos = "";
+#endif
   pw.pw_dir = "/";
   pw.pw_shell = "/bin/sh";
 
diff --git a/gettext-tools/src/msginit.c b/gettext-tools/src/msginit.c
index 228d1d0..d673dee 100644
--- a/gettext-tools/src/msginit.c
+++ b/gettext-tools/src/msginit.c
@@ -1098,7 +1098,7 @@ get_user_pwd ()
 static const char *
 get_user_fullname ()
 {
-#if HAVE_PWD_H
+#if HAVE_PWD_H && !defined(__MVS__)
   struct passwd *pwd = get_user_pwd ();
   if (pwd != NULL)
     {
diff --git a/libtextstyle/examples/color-hello/hello.c b/libtextstyle/examples/color-hello/hello.c
index 5b5221d..e64b65a 100644
--- a/libtextstyle/examples/color-hello/hello.c
+++ b/libtextstyle/examples/color-hello/hello.c
@@ -85,7 +85,11 @@ get_fullname (void)
       struct passwd *pwd = getpwnam (name);
       if (pwd != NULL)
         {
+#ifdef __MVS__
+          const char* gecos = NULL;
+#else
           const char *gecos = pwd->pw_gecos;
+#endif
           if (gecos != NULL)
             {
               /* Use the part before the first comma.
