diff --git a/gettext-tools/autotools/add-to-archive b/gettext-tools/autotools/add-to-archive
index cf96ed3..d9cb55f 100755
--- a/gettext-tools/autotools/add-to-archive
+++ b/gettext-tools/autotools/add-to-archive
@@ -83,7 +83,7 @@ mkdir autopoint-files
 (cd autopoint-files && tar xf ../archive.dir.tar)
 mkdir autopoint-files/$pack_ver
 (cd "$work_archive" && tar cf - .) | (cd autopoint-files/$pack_ver && tar xf -)
-(cd autopoint-files && tar cf ../archive.dir.tar --owner=root --group=root *)
+(cd autopoint-files && tar cf ../archive.dir.tar --owner=root --group=root * && chtag -tcISO8859-1 ../archive.dir.tar)
 
 # Clean up.
 rm -rf autopoint-files
