diff --git a/gettext-tools/libgrep/glthread/lock.c b/gettext-tools/libgrep/glthread/lock.c
index 10af6f2..c78f365 100644
--- a/gettext-tools/libgrep/glthread/lock.c
+++ b/gettext-tools/libgrep/glthread/lock.c
@@ -20,7 +20,10 @@
 #include <config.h>
 
 #include "glthread/lock.h"
-
+#ifdef __MVS__
+  #pragma csect(code, "MULTILOCK#C")
+  #pragma csect(static, "MULTILOCK#S")
+#endif
 /* ========================================================================= */
 
 #if USE_ISOC_THREADS || USE_ISOC_AND_POSIX_THREADS
